File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-0.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        std::stack<int> st;
        for(const string& t:tokens){
            if(t=="+"){
                int a = st.top();
                st.pop();
                int b = st.top();
                st.pop();
                st.push(a+b);
            }
            else if(t=="*"){
                int a = st.top();
                st.pop();
                int b = st.top();
                st.pop();
                st.push(a*b);
            }else if(t=="/"){
                int a = st.top();
                st.pop();
                int b = st.top();
                st.pop();
                st.push(b/a);
            }else if(t=="-"){
                int a = st.top();
                st.pop();
                int b = st.top();
                st.pop();
                st.push(b-a);
            }else{
                st.push(std::stoi(t));
            }
            
        }
        return st.top();
    }
};

File: Data Structures & Algorithms/daily-temperatures/submission-0.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int[] result = new int[temperatures.length];
        Stack<Integer> st = new Stack<>();
        for(int i=0;i<temperatures.length;i++){
            int currTemp = temperatures[i];
            while(!st.isEmpty() && currTemp > temperatures[st.peek()]){
                int pastIndex = st.pop();
                result[pastIndex] = i - pastIndex;
            }
            st.push(i);
        }
        return result;
    }
}
